light-system-programs/tests/merkle_tree_tests.ts

```diff
@@ -1054,4 +1060,128 @@ describe("Merkle Tree Tests", () => {
+
+    const newMerkleTreePubkey = MerkleTreeConfig.getTransactionMerkleTreePda(
+      new anchor.BN(1),
+    );
+
+    let txParams = new TransactionParameters({
+      outputUtxos: [shieldUtxo],
+      eventMerkleTreePubkey: MerkleTreeConfig.getEventMerkleTreePda(),
+      transactionMerkleTreePubkey:
+        MerkleTreeConfig.getTransactionMerkleTreePda(),
+      senderSpl: userTokenAccount,
+      senderSol: ADMIN_AUTH_KEYPAIR.publicKey,
+      action: Action.SHIELD,
+      poseidon: POSEIDON,
+      verifierIdl: IDL_VERIFIER_PROGRAM_ZERO,
+    });
+    let transaction = new Transaction({
+      provider: lightProvider,
+      params: txParams,
+    });
+    transaction.remainingAccounts!.nextTransactionMerkleTree = {
+      isSigner: false,
+      isWritable: true,
+      pubkey: newMerkleTreePubkey,
+    };
+
+    await transaction.compileAndProve();
+    await transaction.sendAndConfirmTransaction();
+
+    let leavesPdas = await SolMerkleTree.getUninsertedLeavesRelayer(
+      newMerkleTreePubkey,
+    );
+    const merkleTreeUpdateState = solana.PublicKey.findProgramAddressSync(
+      [
+        ADMIN_AUTH_KEYPAIR.publicKey.toBytes(),
+        anchor.utils.bytes.utf8.encode("storage"),
+      ],
+      merkleTreeProgram.programId,
+    )[0];
+
+    await merkleTreeProgram.methods
+      .initializeMerkleTreeUpdateState()
+      .accounts({
+        authority: ADMIN_AUTH_KEYPAIR.publicKey,
+        merkleTreeUpdateState: merkleTreeUpdateState,
+        systemProgram: SystemProgram.programId,
+        rent: DEFAULT_PROGRAMS.rent,
+        transactionMerkleTree: newMerkleTreePubkey,
+      })
+      .remainingAccounts([leavesPdas[0]])
+      .preInstructions([
+        solana.ComputeBudgetProgram.setComputeUnitLimit({ units: 1_400_00 }),
+      ])
+      .signers([ADMIN_AUTH_KEYPAIR])
+      .rpc(confirmConfig);
+
+    await executeMerkleTreeUpdateTransactions({
+      signer: ADMIN_AUTH_KEYPAIR,
+      merkleTreeProgram,
+      transactionMerkleTree: newMerkleTreePubkey,
+      connection: provider.connection,
+      merkleTreeUpdateState,
+    });
+
+    await checkMerkleTreeUpdateStateCreated({
+      connection: provider.connection,
+      merkleTreeUpdateState,
+      transactionMerkleTree: newMerkleTreePubkey,
+      relayer: ADMIN_AUTH_KEYPAIR.publicKey,
+      leavesPdas: [leavesPdas[0]],
+      current_instruction_index: 56,
+      merkleTreeProgram,
+    });
+
+    let newMerkleTreeAccountPrior =
+      await merkleTreeProgram.account.transactionMerkleTree.fetch(
+        newMerkleTreePubkey,
+      );
+
+    await merkleTreeProgram.methods
+      .insertRootMerkleTree(new anchor.BN(254))
+      .accounts({
+        authority: ADMIN_AUTH_KEYPAIR.publicKey,
+        merkleTreeUpdateState: merkleTreeUpdateState,
+        transactionMerkleTree: newMerkleTreePubkey,
+        logWrapper: SPL_NOOP_ADDRESS,
+      })
+      .signers([ADMIN_AUTH_KEYPAIR])
+      .rpc(confirmConfig);
+
+    await checkMerkleTreeBatchUpdateSuccess({
```

Comment:
I think we could simplify the merkle tree update since this check in line 1177 is the only one we care about after sendAndConfirmTransaction.